### ISSUE TITLE
onValueChange Event triggers repeatedly

### ISSUE THREAD

**Issue by Mangirish-Bhagat** [hossein-zare/react-native-dropdown-picker#371]
<DropDownPicker
                style={{ borderRadius: 10 }}
                open={this.state.open}
                value={this.state.value}
                items={this.props.order.dishdet}
                setOpen={(open)=>{this.opendrop(open)}}
                setValue={this.setValue}

              onChangeValue = {(value) => {
                this.setState({value: value})
              this.getPlaylistMedia();
      }}
    />

so basically **onChangeValue** event get triggered countless times and then there is a warning :

**Warning: Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.**

And then the drop down stops working, it does not open

**Comment by TeyimPila** [comment]
I am experiencing this too

**Comment by kelvinaliyanto** [comment]
any solution to this issue?

**Comment by mikehardy** [comment]
I haven't experienced this, sorry. The message seems pretty explicit though - I basically copy pasted from the example and everything worked. The issue description here is not a complete example (https://stackoverflow.com/help/minimal-reproducible-example) so I can't run it and make any coherent comment about why it wouldn't work.

I would try a simple App.js based on the example using the useState hooks to verify that works then build up your dropdown solution from there again, if I were troubleshooting this.

**Comment by DeshanGit** [comment]
items={props.stationListDataState.stationList}

Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.

how to fix this issue? any solution ??

**Comment by mikehardy** [comment]
> but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.

Heed the message, add a dependency array or fix it, so that when you setState in useEffect, useEffect understands which state variables are in play so it doesn't loop. There is plenty of reading on useEffect correct usage in medium articles etc

**Comment by Devanshu008** [comment]
I got same this error when i set a value using onChangeValue. 
any Solution?

**Comment by mikehardy** [comment]
So, if you have a useEffect that runs after a re-render, and that useEffect triggers a state change, which triggers a re-render, you get an infinite loop...

This is expected, isn't it?

useEffect must have a dependency array that includes the state values it is responding too, so it only runs once. That's react-native useEffects most important thing.

This is not a bug in the library it is a bug in project-specific code, there is nothing actionable here so I am going to close and lock this.

Everyone suffering from this (it happened to me too! I understand) needs to carefully examine your useEffect/state interaction and fix your useEffect hook
